### ISSUE TITLE
[feature] Default CMAKE_POSITION_INDEPENDENT_CODE to OFF in CMakeToolchain when building for bare-metal targets

### ISSUE THREAD

**Issue by jwillikers** [conan-io/conan#9791]
### Environment Details (include every applicable attribute)
  * Operating System+version: Fedora Silverblue 35
  * Compiler+version: GCC 11
  * Conan version: 1.41.0
  * Python version: 3.10

Position independent code is not always supported for bare-metal targets.
Specifically for Arm targets, PIE support is deprecated for the GNU Arm Embedded compiler.
Enabling PIE can be problematic when building for these targets and cause unexpected crashes.
The CMakeToolchain generator should account for this by defaulting `CMAKE_POSITION_INDEPENDENT_CODE` to `OFF` for bare metal targets.

Reference: https://developer.arm.com/documentation/100748/0612/mapping-code-and-data-to-the-target/bare-metal-position-independent-executables

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
I think this maybe should be intercepted before and remove the option. 

Setting CMAKE_POSITION_INDEPENDENT_CODE only happens if fPIC is defined. If not defined, Conan will do nothing, which is OFF, and it should be ok?

If the option fPIC is defined, the issue is that it would be generating 2 different binaries for the 2 different values of fPIC, while the binary would be identical, because CMakeToolchain would always force the variable to OFF irrespective of the current ``self.options.fPIC``. Seems like in this case, the option ``fPIC`` should be removed completely from the options?

**Comment by jwillikers** [comment]
Yes, that sounds like a good idea so long as it is enforceable on dependencies. Consider `fmt` being used as a dependency for a project which does not define `fPIC`. Would `fmt` be built with `fPIC` enabled still? That would be problematic for most bare-metal configurations.

If an package can be built for both bare metal and standard operating systems, the latter which might use `fPIC`, would you just delete the option like is commonly done for Windows?

**Comment by memsharded** [comment]
Yes, I think we need some kind of generalized handling of fPIC, at the moment is something that has to be manually added to all recipes to remove it for the Windows case. So maybe it is time to generalize it in something like:

```python
def config_options(self):
      handle_fpic(self)
```
To provide an easy implementation, while retaining complete user-configurability. Towards 2.0 we could consider calling such function implicitly and allowing to skip it as an opt-out, if we favor recipe simplicity of being explicit. This is indeed a very valid feature.
